<launch>
  <arg name="approve" default="no" />
  <arg name="web3_http_provider" default="http://127.0.0.1:8545" />
  <arg name="ipfs_http_provider" default="http://127.0.0.1:5001" />

  <node name="agent" pkg="iiot_data_secure" type="blockchainization_agent_node" output="screen">
    <param name="ipfs_http_provider" value="$(arg ipfs_http_provider)" />
    <param name="web3_http_provider" value="$(arg web3_http_provider)" />
    <param name="model" value="QmZRNkDtUzGSfSN5XKCXaMd6tLcY5TJLLrhpToWhauCfJh"/>
 <!--    <param name="token" value="0x3cBAF1d511Adf5098511B5c5B39e1F1b506C1AFE" /> XRT -->
    <param name="token" value="0xc34B089D3Dfe7bdf39a7Cd3D64116b2395dba845" /> <!-- DEUT -->
    <param name="liability_factory" value="0x44CFBcb1Ca0d3df0925dDA3354E955d38d78ad6B" />
    <param name="bid_lifetime" value="100" /> <!-- Ethereum blocks, ~12-15 sec for one -->
    <param name="approve" value="$(arg approve)" />
  </node>
  
  <include file="$(find iiot_data_secure)/launch/objective_composer.launch" />
</launch>
